pages/_app.tsx

```diff
@@ -1,35 +1,43 @@
-import '../styles/global.css';
+import type { AppProps } from "next/app";
+import React, { useEffect, useState } from "react";
 
-import React, { useState, useEffect } from 'react';
-import { AppContextProvider } from '../components/appContext';
+import type { Book_t } from "../components/appContext";
+import { AppContextProvider } from "../components/appContext";
+import "../styles/global.css";
 
-function MyApp({ Component, pageProps }) {
-  const [favorites, setFavorites] = useState([]);
+function MyApp({ Component, pageProps }: AppProps) {
+  const [favorites, setFavorites] = useState<Book_t[]>([]);
 
   useEffect(() => {
-    const savedFavorites = JSON.parse(localStorage.getItem('favorites')) || [];
+    const savedFavoritesJSON = localStorage.getItem("favorites");
+    const savedFavorites =
+      savedFavoritesJSON !== null
+        ? (JSON.parse(savedFavoritesJSON) as Book_t[])
+        : [];
     setFavorites(savedFavorites);
   }, []);
 
   useEffect(() => {
-    localStorage.setItem('favorites', JSON.stringify(favorites));
+    localStorage.setItem("favorites", JSON.stringify(favorites));
   }, [favorites]);
 
-  const handleFavorite = (book) => {
+  const handleFavorite = (book: Book_t) => {
     setFavorites((prevFavorites) => {
-      const isAlreadyFavorite = prevFavorites.some(f => f.id === book.id);
+      const isAlreadyFavorite = prevFavorites.some(
+        (f: Book_t) => f.id === book.id,
+      );
```

Comment:
nie potrzebujesz tu adnotacji
```suggestion
      const isAlreadyFavorite = prevFavorites.some(
        (f) => f.id === book.id,
      );
```